projects/swimlane/ngx-ui/src/lib/components/dropdown/dropdown-menu.directive.ts

```diff
@@ -1,15 +1,25 @@
-import { ElementRef, Directive } from '@angular/core';
+import { ElementRef, Directive, Inject, PLATFORM_ID, EventEmitter } from '@angular/core';
+import { InViewportDirective, InViewportService } from 'ng-in-viewport';
 
 @Directive({
   // tslint:disable-next-line:directive-selector
   exportAs: 'ngxDropdownMenu',
   selector: 'ngx-dropdown-menu',
   host: { class: 'ngx-dropdown-menu' }
 })
-export class DropdownMenuDirective {
+export class DropdownMenuDirective extends InViewportDirective {
```

Comment:
Is there a way to do this without inheritance?  Maybe using the service directly?

Reply:
The best way I can figure think of to achieve the same result is by creating a separate directive that follows these patterns. They also involve inheritance but not in the same way as we are doing it now.

I'd have to try. But using the directive is a must, I'm afraid.

Reply:
Is the Overlay from Angular CDK an option?

Reply:
Why not use it like below without inheritance.
https://github.com/k3nsei/ng-in-viewport/blob/develop/projects/example/src/app/page-lazy-images/page-lazy-images.component.html#L8
https://github.com/k3nsei/ng-in-viewport/blob/develop/projects/example/src/app/page-lazy-images/lazy-img.directive.ts#L61